mantidproject/mantid
# Fix picking up ROI Detector IDs from lookup table in preview reduction

**Description of work**

**Purpose of work**

This PR ensure that ROI Detector IDs that have been entered into the lookup table on the ISIS Reflectometry interface Experiment Settings tab are correctly applied to both the preview reduction and sum banks steps when nothing has been selected on the instrument view plot.



**Summary of work**

This PR makes changes in `RowProcessingAlgorithm.cpp` to ensure that the ROI detector IDs from the lookup table are applied both in a normal batch reduction and a preview reduction. I have made a change in the `PreviewPresenter` so that when we delete a selected detector region from the instrument view plot on the preview tab we correctly clear the selected banks in the `PreviewModel`. This prevents us failing to use lookup table values after deleting the region selection.

On the preview tab, it should also be the case that the slice viewer plot will only display the banks as summed if there is a detector ROI available (either from the instrument view selection or the experiment settings lookup table). In the `PreviewPresenter`, changes have been made to ensure that we check the lookup table as well as the instrument viewer selection when checking whether to sum the input workspace for the slice viewer plot.


**To test:**


Run through the instructions on the linked issue and test that the reduction now completes successfully and that the slice viewer plot on the preview tab shows summed data (there is one region of data rather than four). To see this clearly choose "SymmetricLog10" from the dropdown underneath the colour bar.

This is the basic functionality, but also check that when a region is selected from the instrument view plot the reduction uses that instead. Check that if all regions are deleted from the instrument view then any information entered into the experiment settings lookup table is used (note that the lookup row will only be picked up if it has no angle information or if the angle for the row matches what is on the preview tab).

Fixes #35911. 






---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
